tests/notebooks/e2e-wine/e2e-wine-kfp-mlflow-seldon.ipynb

```diff
@@ -0,0 +1,1233 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "25c156ad-21a9-4ba8-86fb-d149f71dceca",
+   "metadata": {},
+   "source": [
+    "### Hit SeldonDeployment for Predictions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 32,
+   "id": "c358116b-609f-410a-96a5-2455f9764336",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2023-08-09T14:59:51.933332Z",
+     "iopub.status.busy": "2023-08-09T14:59:51.932943Z",
+     "iopub.status.idle": "2023-08-09T14:59:51.946664Z",
+     "shell.execute_reply": "2023-08-09T14:59:51.945430Z"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "SeldonDeployment URL: http://kf-testing-wine-model.test.svc.cluster.local:8000/api/v1.0/predictions\n"
+     ]
+    }
+   ],
+   "source": [
+    "sd = get_seldon_deployment(SELDON_DEPLOYMENT_NAME, NAMESPACE)\n",
+    "url = sd['status']['address']['url']\n",
+    "print(\"SeldonDeployment URL:\", url)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 33,
+   "id": "dc9c3b09-2f2f-4549-b0e5-458e9762fc93",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2023-08-09T14:59:51.950885Z",
+     "iopub.status.busy": "2023-08-09T14:59:51.949876Z",
+     "iopub.status.idle": "2023-08-09T14:59:52.009876Z",
+     "shell.execute_reply": "2023-08-09T14:59:52.008708Z"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{\"data\":{\"names\":[],\"ndarray\":[5.737135502528464]},\"meta\":{\"requestPath\":{\"classifier\":\"seldonio/mlflowserver:1.17.0\"}}}\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "inference_input = {\n",
+    "  \"data\": {\n",
+    "      \"ndarray\": [\n",
+    "          [\n",
+    "              10.1, 0.37, 0.34, 2.4, 0.085, 5.0, 17.0, 0.99683, 3.17, 0.65, 10.6\n",
+    "          ]\n",
+    "      ]\n",
+    "  }\n",
+    "}\n",
+    "response = requests.post(url, json=inference_input)\n",
+    "print(response.text)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 34,
+   "id": "e3d8af0d-7fb1-466d-acb4-e1c2bf4eb05a",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2023-08-09T14:59:52.021717Z",
+     "iopub.status.busy": "2023-08-09T14:59:52.016340Z",
+     "iopub.status.idle": "2023-08-09T14:59:52.026727Z",
+     "shell.execute_reply": "2023-08-09T14:59:52.025730Z"
+    },
+    "tags": [
+     "raises-exception"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "res = response.json()\n",
+    "# verify that the predictions are as expected\n",
+    "assert res.get(\"data\") and res.get(\"data\").get(\"ndarray\"), \"Failed to get predictions!\"\n",
+    "predictions = res[\"data\"][\"ndarray\"]\n",
+    "assert len(predictions) == 1, \"Predictions not in the expected format!\"\n",
+    "assert predictions == [5.737135502528464], \"Predictions different than expected!\""
```

Comment:
Probably yes, but can we trust this prediction will have the same value 100% of the times? What is the information we are obtaining from this assertion?

Reply:
I ran this many times and it succeeded consistently, but it's true that I haven't looked into how this is created. However, since we use a seed for the model training I'm pretty confident that the predictions will be predictable.

Reply:
Oh, alright, as long as we don't change the seed, we should be good.